apps/sched/interface.html

```diff
@@ -88,29 +88,62 @@
 function upload() {
   Util.showModal("Saving...");
   Util.writeStorage("sched.json", JSON.stringify(alarms), () => {
-    location.reload(); // reload so we see current data
+    Puck.write(`\x10load()\n`, () => { // reload watch to load alarms
+      setTimeout(() => {
```

Comment:
I'm not a fan of this, but the introduction of `Puck.write` / `load()` means that when we reload the interface and [re-read `sched.json`](https://github.com/espruino/BangleApps/blob/825594dc9fa0ba0af5803805d0e2ff3e3fcf15f7/apps/sched/interface.html#L252-L253), I get this data back:

```
×]<[...]
   ^^^^^ sched.json contents
^^^ unknown
```

Unless I wait for a little bit. Is there something I might be missing?

Reply:
Ahh - it's possible that when you reload you see the data that's meant to be sent to gadgetbridge? Or maybe just the Espruino header text? Doing `Puck.debug=3` might help you see more of what's happening but as you say about all you can really do is delay...

Is there a reason to force reload? In most cases we reset the Bangle and then just display that 'press the button to reload' message which at least helps us avoid this sort of thing

Reply:
Yes perhaps that's it - I see the below output with `Puck.debug` set to 3, not sure if there's much of a clue there, maybe the `NaN` size? But yes - I prefer matching the existing "Press button to reload" approach, I'll adjust it to do that. Thanks!

<details>
<summary>

console output with `Puck.debug=3`

</summary>

```
<COMMS> writeFile "sched.json"
comms.js:29 <COMMS> getProgressCmd START
comms.js:29 <COMMS> getProgressCmd 100%
comms.js:83 <COMMS> Response:  "OK\r"
index.js:214 IFRAME loaded
interface.js:104 <INTERFACE> init message received {id: 'BANGLEJS2', version: '2v17', info: {…}, connected: true, appsInstalled: Array(47), …}
comms.js:480 <COMMS> readFile "sched.json"
comms.js:473 <COMMS> readTextBlock read started...
comms.js:459 <COMMS> size is NaN
VM2336:1 Uncaught SyntaxError: Unexpected token '×', "×]8[{"t":3"... is not valid JSON
    at JSON.parse (<anonymous>)
    at interface.html:257:19
    at interface.js:116:13
(anonymous) @ interface.html:257
(anonymous) @ interface.js:116
postMessage (async)
(anonymous) @ index.js:248
Promise.then (async)
(anonymous) @ index.js:247
postMessage (async)
readStorage @ interface.js:48
getData @ interface.html:256
onInit @ interface.html:282
(anonymous) @ interface.js:108
postMessage (async)
(anonymous) @ index.js:266
load (async)
iframeSetup @ index.js:213
(anonymous) @ index.js:353
handleAppInterface @ index.js:337
(anonymous) @ index.js:895
```

</details>

Reply:
Ahh - try Puck.debug=3 when you don't have interface.html open - I think you might have been running it in interface.html's JS instance

Reply:
Nice call, that's done it - looks like it's the powermanager and popcon apps I have perhaps delaying the response.

There's no more exception although it does hang now - but I guess we can sidestep this with the "Hold button to reload" message?

```
<COMMS> readFile "sched.json"
<BLE> Sending "\u0003\u0010(function() {\nvar s = require(\"Storage\").read(\"sched.json\");\nBluetooth.println(((s.length+2)/3)<<2); // estimate file size\nfor"
<BLE> Sent
<BLE> Sending " (var i=0;i<s.length;i+=384) Bluetooth.print(btoa(s.substr(i,384)));\nBluetooth.print(\"ÿ\");\n})()\n"
<BLE> Sent
<COMMS> readTextBlock read started...
<BLE> Received "\r\u001b[JExecution Interrupted\r\nat line 25 col 1295 in .boot0\r\n...ldRead(fname,skipExceptions);if(j&&/\\.info$/.test(fname)){va...\r\n  "
<BLE> Received "                            ^\r\nin function \"readJSON\" called from line 26 col 278 in .boot0\r\n...N(\"powermanager.def.json\",true)|"
<BLE> Received "|{};if(!def.start)def.start=Date....\r\n                                   ^\r\nin function called from line 26 col 3603 in .boot0\r\n"
<BLE> Received "...rging)chargeStart=undefined;});}})();;\r\n                                      ^\r\n>1104\r\nW3sidCI6MzAxMjAwMDAsImRlbCI6ZmFsc2UsI"

# there's then a bunch of atob()'d data after this, but no unable-to-parse-json error now
```